Sources/Datadog/RUM/AutoInstrumentation/Views/UIKitRUMViewsHandler.swift

```diff
@@ -29,28 +37,50 @@ internal class UIKitRUMViewsHandler: UIKitRUMViewsHandlerType {
         self.subscriber = commandsSubscriber
     }
 
-    func notify_viewWillAppear(viewController: UIViewController, animated: Bool) {
+    func notify_viewDidAppear(viewController: UIViewController, animated: Bool) {
         if let rumView = predicate.rumView(for: viewController) {
-            subscriber?.process(
-                command: RUMStartViewCommand(
-                    time: dateProvider.currentDate(),
-                    identity: viewController,
-                    path: rumView.path,
-                    attributes: rumView.attributes
-                )
-            )
+            startIfNotStarted(rumView: rumView, for: viewController)
         }
     }
 
-    func notify_viewWillDisappear(viewController: UIViewController, animated: Bool) {
-        if let rumView = predicate.rumView(for: viewController) {
+    func notify_viewDidDisappear(viewController: UIViewController, animated: Bool) {
+        guard let topViewController = inspector.topViewController() else {
+            return
+        }
+
+        if let rumView = predicate.rumView(for: topViewController) {
+            startIfNotStarted(rumView: rumView, for: topViewController)
+        }
+    }
+
+    // MARK: - Private
+
+    private weak var lastStartedViewController: UIViewController?
+
+    private func startIfNotStarted(rumView: RUMViewFromPredicate, for viewController: UIViewController) {
```

Comment:
If the same view controller is both auto instrumented and started manually, this will lead to duplicating the RUM View - first one will be finished and the second one will be started.

I think this is correct, as having single `UIViewController` both auto- and manually- instrumented is the same wrong as starting the view twice in manual instrumentation. Our configuration gives user enough flexibility to opt-out from auto instrumenting particular views and managing them manually.